govcd/api.go

```diff
@@ -178,3 +178,100 @@ func checkResp(resp *http.Response, err error) (*http.Response, error) {
 		return nil, fmt.Errorf("unhandled API response, please report this issue, status code: %s", resp.Status)
 	}
 }
+
+// Helper function creates request, run it, check response and parse task from response.
+// pathUlr - request URL
+// requestType - HTTP method type
+// contentType - value which will be set for "Content-Type"
+// errorMessage - error message which to return when error happen
+// payload - XML struct which will be marshalled and added as body/payload
+// E.g. client.ExecuteTaskRequest(updateDiskLink.HREF, http.MethodPut, updateDiskLink.Type, "error update disk: %s", xmlPayload)
+func (client *Client) ExecuteTaskRequest(pathURL, requestType, contentType, errorMessage string, payload interface{}) (Task, error) {
```

Comment:
I don't agree on this, I changed code with this function and a lot of places we return directly responses from this function and also didn't see need for response. When some developing govcd will see need function with direct handling of response - he can add it, I don't like guessing future.

Reply:
This is not a matter of future needs. The response should be returned, so that the caller can have extra information that is not contained in either `Task` or `error`.
Adding the response won't undermine the compactness of the current implementation, but it would give access to information that could be vital. If I need it for debugging, I'd have to rewrite the calling code.

Reply:
I added response for ExecuteRequest, this seems only one func reasonable for that. Though didn't like that a lot of `_,` had to add everywhere.